src/esp/sim/Simulator.h

```diff
@@ -65,6 +65,14 @@ class Simulator {
   explicit Simulator(const SimulatorConfiguration& cfg);
   virtual ~Simulator();
 
+  /**
+   * @brief Closes the simulator and frees all loaded assets and GPU contexts
+   *
+   * @warning Must reset the simulator to its "just after constructor" state for
+   * python inheritance to function correctly
```

Comment:
Can we add more context here, what should be added here: each shared pointer member or etc and what will happen if that won't be done.